consensus/src/pow/work.rs

```diff
@@ -0,0 +1,133 @@
+#![allow(dead_code)]
+
+use crate::pow::helpers::{
+    calculate_new_target, check_difficulty, is_for_retarget, retarget_block_time, special_rules,
+};
+use crate::pow::temp::BlockIndex;
+use crate::pow::{Error, PoW};
+use common::chain::block::Block;
+use common::primitives::consensus_data::{ConsensusData, PoWData};
+use common::primitives::{Compact, Idable};
+use common::Uint256;
+
+pub fn check_proof_of_work(hash: Uint256, bits: Compact) -> bool {
+    if let Ok(target) = Uint256::try_from(bits) {
+        hash > target
```

Comment:
Two things:

1. This should be the other way around. The hash has to be less than the target.
2. Depending on how this is gonna be used (it's never used anywhere so far), if this is gonna be used once per block, then this is OK. If this is gonna be used for mining, then this is bad, because we don't want to reinterpret the bits every time we change the nonce.

Reply:
Yes, that's wrong. :( I will change it asap.